vhost-device-spi/src/vhu_spi.rs

```diff
@@ -0,0 +1,1206 @@
+            let mem = atomic_mem.memory();
+
+            let mut writter = desc_chain
+                .clone()
+                .writer(&mem)
+                .map_err(|_| Error::DescriptorReadFailed)?;
+
+            if writter.available_bytes() > size_of::<u8>() {
+                let rx_len = (writter.available_bytes() - size_of::<u8>()) as u32;
+                let mut writter_status = writter
+                    .split_at(rx_len as usize)
+                    .map_err(|_| Error::DescriptorReadFailed)?;
+
+                writter_status
+                    .write_obj(in_hdr)
+                    .map_err(|_| Error::DescriptorWriteFailed)?;
+
+                writter
+                    .write(&reqs[i].rx_buf)
+                    .map_err(|_| Error::DescriptorWriteFailed)?;
+
+                len += rx_len;
+            } else {
+                writter
+                    .write_obj(in_hdr)
+                    .map_err(|_| Error::DescriptorWriteFailed)?;
+            }
+
+            // Write the transfer status
+            if vring.add_used(desc_chain.head_index(), len).is_err() {
+                warn!("Couldn't return used descriptors to the ring");
+            }
+        }
+
+        Ok(true)
+    }
+
+    /// Process the requests in the vring and dispatch replies
+    fn process_queue(&self, vring: &VringRwLock) -> Result<()> {
+        let Some(ref atomic_mem) = self.mem else {
+            return Err(Error::NoMemoryConfigured);
+        };
+        let requests: Vec<_> = vring
+            .get_mut()
+            .get_queue_mut()
+            .iter(atomic_mem.memory())
+            .map_err(|_| Error::DescriptorNotFound)?
+            .collect();
+
+        if self.process_requests(requests, vring)? {
+            // Send notification once all the requests are processed
+            vring
+                .signal_used_queue()
+                .map_err(|_| Error::NotificationFailed)?;
+        }
+
+        Ok(())
+    }
+}
+
+/// VhostUserBackendMut trait methods
+impl<D: 'static + SpiDevice + Sync + Send> VhostUserBackendMut for VhostUserSpiBackend<D> {
+    type Vring = VringRwLock;
+    type Bitmap = ();
+
+    fn num_queues(&self) -> usize {
+        NUM_QUEUES
+    }
+
+    fn max_queue_size(&self) -> usize {
+        QUEUE_SIZE
+    }
+
+    fn features(&self) -> u64 {
+        // this matches the current libvhost defaults except VHOST_F_LOG_ALL
+        1 << VIRTIO_F_VERSION_1
+            | 1 << VIRTIO_F_NOTIFY_ON_EMPTY
+            | 1 << VIRTIO_RING_F_INDIRECT_DESC
+            | 1 << VIRTIO_RING_F_EVENT_IDX
+            | VhostUserVirtioFeatures::PROTOCOL_FEATURES.bits()
+    }
+
+    fn protocol_features(&self) -> VhostUserProtocolFeatures {
+        VhostUserProtocolFeatures::MQ
+            | VhostUserProtocolFeatures::CONFIG
+            | VhostUserProtocolFeatures::REPLY_ACK
+    }
+
+    fn set_event_idx(&mut self, enabled: bool) {
+        dbg!(self.event_idx = enabled);
```

Comment:
```suggestion
        self.event_idx = enabled;
```

The value of a statement is unit `()` so this is not useful at all.

Reply:
OK, I will remove this line.